lib/components/Selection control/Checkbox/index.tsx

```diff
@@ -0,0 +1,180 @@
+            }
+        };
+
+        const uncheckedIcon = is_indeterminate ? (
+            <StandaloneSquareMinusFillIcon
+                iconSize="sm"
+                className={clsx("quill-checkbox__icon", {
+                    "quill-checkbox__icon--disabled": disabled,
+                })}
+            />
+        ) : (
+            <StandaloneSquareBoldIcon
+                iconSize="sm"
+                className={clsx(
+                    "quill-checkbox__icon",
+                    "quill-checkbox__icon--default",
+                    {
+                        "quill-checkbox__icon--disabled": disabled,
+                    },
+                )}
+            />
+        );
+
+        const infoIcon =
+            size === "sm" ? (
+                <LabelPairedCircleInfoSmRegularIcon
+                    className={clsx(
+                        "quill-checkbox__info-icon",
+                        {
+                            "quill-checkbox__info-icon--disabled": disabled,
+                        },
+                        infoIconClassName,
+                    )}
+                />
+            ) : (
+                <LabelPairedCircleInfoMdRegularIcon
+                    className={clsx(
+                        "quill-checkbox__info-icon",
+                        {
+                            "quill-checkbox__info-icon--disabled": disabled,
+                        },
+                        infoIconClassName,
+                    )}
+                />
+            );
+
+        return (
+            <div className={clsx("quill-checkbox", wrapperClassName)}>
+                <div className="quill-checkbox__wrapper">
+                    <input
+                        id={rest.id ?? name}
+                        className="quill-checkbox__box"
+                        type="checkbox"
+                        checked={is_checked}
+                        disabled={disabled}
+                        ref={ref}
+                        name={name}
+                        onChange={onInputChange}
+                        onKeyDown={onKeyDown}
+                        {...rest}
+                    />
+                    {is_checked ? (
+                        <StandaloneSquareCheckFillIcon
+                            iconSize="sm"
+                            className={clsx("quill-checkbox__icon", {
+                                "quill-checkbox__icon--disabled": disabled,
+                            })}
+                        />
+                    ) : (
+                        uncheckedIcon
+                    )}
+                </div>
+                <label htmlFor={rest.id ?? name}>
+                    <Text
+                        size={size}
+                        as="span"
+                        className={clsx(
+                            "quill-checkbox__label",
+                            {
+                                "quill-checkbox__label--disabled": disabled,
```

Comment:
can we apply opacity to your component instead of changing the text color (and icon color above) for disabled state?

Reply:
The reason is that currently only -semantic-color exist in both light and dark mode. If I used color, then it would not work for dark theme. 
Should I leave code like this or return back to usage var(--core-color-opacity-black-300) and etc but without dark theme?